### ISSUE TITLE
0.7 EXP version crashing on Mac in Ableton and Logic

### ISSUE THREAD

**Issue by jacbou** [publicsamples/Xolotl#2]
Plugin crashes Ableton and Logic
I had the previous version 0.6 installed before and that was working. Are there any old files or configuration files that I have to or can remove to start with a blank install to test this? (I deleted /Users/USER/Library/Application Support/modularsamples/Xolotl as well plus the VST3 and Component versions in their folders and did an install again, but still crashing.

**Comment by RickTaylor79** [comment]
Hmm, seems fine on logic here for the most part, not tested Ableton yet. 

Generally speaking it's a good idea to remove old content with these things but it looks like you have that covered..

Are there any particular circumstances that lead up to the crash? Do you have the samples installed?

**Comment by jacbou** [comment]
No, nothing special. Quetzalcoatl and Oi Grandad work fine. It's an intel mac. I have Samples installed on an external HD, but it does not even select me where to look for the samples (crashes instantly). I now set the samplesfound to 0 in general config and restarted. It crashed but before created a LinkOSX file with a directoy name. I chnaged that to the real sample folder. Still crashes. Maybe just compile a new version or look at the compiler settings? I also tried the free version from github. also crashes.

**Comment by jacbou** [comment]
OS is MacOS latest version (15.1) on Intel. I' mainly using the VST3 version. I don't know how Ableton logs plugin actions, but could find this in Ardour's plugin scanner (shows the plugin as "error" after scnanning:

VST3 module-path '/Library/Audio/Plug-Ins/VST3/Xolotl.vst3/Contents/Info.plist'
[Info]: Scanning: /Library/Audio/Plug-Ins/VST3/Xolotl.vst3
Error: signal 11
 ---8<---
  2   libsystem_platform.dylib            0x00007ff80fd8ae9d _sigtramp + 29
  3   ???                                 0x00000000000002ff 0x0 + 767
  4   Xolotl                              0x000000010aca44aa GetPluginFactory + 1523146
  5   Xolotl                              0x000000010af663f0 GetPluginFactory + 4414736
  6   Xolotl                              0x000000010af69c2b GetPluginFactory + 4429131
  7   Xolotl                              0x000000010adc1bd5 GetPluginFactory + 2692341
  8   Xolotl                              0x000000010adc5b28 GetPluginFactory + 2708552
  9   Xolotl                              0x000000010adfa8bd GetPluginFactory + 2925021
  10  Xolotl                              0x000000010ad55446 GetPluginFactory + 2248038
  11  Xolotl                              0x000000010ab3be3b GetPluginFactory + 46939
  12  ardour-vst3-scanner                 0x0000000105936f03 ardour-vst3-scanner + 36611
  13  ardour-vst3-scanner                 0x000000010593e459 ardour-vst3-scanner + 66649
  14  dyld                                0x00007ff80f9c82cd start + 1805
 --->8---

Don't know, if this is of any help?

**Comment by jacbou** [comment]
Just tried on my older MacBook with Big Sur. Same problem. Both had Xolotl 0.6 installed before.

**Comment by RickTaylor79** [comment]
Okay, that rules out my theory (if in doubt blame the new OS). Thanks for the crash report, between this a few other clues gives me something to work with..

The most significant difference between v0.7 and 0.6, Oi, Grandad! and Quetzalcoatl is that it uses FAUST for some of the DSP.  I just don't understand why it works perfectly well for me (m3 air Sonoma), but not you on two separate machines.

 I suppose the difference in architecture could something to do with it. The plugin _is_ universal binary, though. I'm gonna fire up Sequoia in a VM, and see what happens. 

Another quick question, in Logic does the plugin actually validate after a reset and rescan?

**Comment by RickTaylor79** [comment]
Okay, I've uploaded a potential fix, I can't guarantee it will work, please let me know how you get on with it. If it works I'll update the expanded version:

https://github.com/publicsamples/Xolotl/releases/download/0.7.0/Xolotl.Mac.Fix.zip

**Comment by jacbou** [comment]
Hi just tested also on the other MacBook. Works fine!

**Comment by jacbou** [comment]
Tested only VST3. In Logic this error occurs dureing validation:
PUBLISHED PARAMETER INFO:

# # # 80 Global Scope Parameters:
ERROR: Parameter values are different since last set - probable cause: a Meta Param Flag is NOT set on a parameter that will change values of other parameters.

* * FAIL
--------------------------------------------------
AU VALIDATION FAILED: CORRECT THE ERRORS ABOVE.

**Comment by RickTaylor79** [comment]
Okay, thanks for the reports. I've posted fixed versions to both the [release page here](https://github.com/publicsamples/Xolotl/releases/tag/0.7.0), and the expanded version over on [Gumroad](https://app.gumroad.com/library).
